receiver/otlpreceiver/otlp.go

```diff
@@ -50,6 +54,39 @@ type otlpReceiver struct {
 	startServerOnce sync.Once
 }
 
+// httpBodyCompressionHandler is a HTTP middleware checks the "Content-Encoding" HTTP header and if
+// a compression such as "gzip", "deflate", "zlib", is found, the request body will
+// be uncompressed accordingly. It it encounters an error while do is, the original request is passed untouched.
+func httpBodyCompressionHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.Header.Get("Content-Encoding") {
+		case "gzip":
+			gr, err := gzip.NewReader(r.Body)
+			if err != nil {
+				break
```

Comment:
This appears to silently ignore the errors. I think we need to return "400 Bad Request" to the client (I think all errors here are an indication of the bad data from the client, if I am wrong then we may need to return a different HTTP status code). Please also cover the error cases in the tests.

Reply:
That sounds reasonable. I will make the changes to address this. thanks!